### PR TITLE
Expand provider capabilities

### PR DESCRIPTION
**Pull request by Melchizedech**
- Remove useless (debug) logs
- rename Waas object to WaasContainer
- added waasNetwork object (to handle network configuration)
- added OPA admission rules
- added defenderSettings 
- added scanSettings
-

### CODE REVIEW

File: sdk/settings/defender.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	}
+
+	settings := DefenderSettings{}
+
+	_, err = c.Do(req, &unpacker)
+	if err != nil {
+		return nil, err
+	}
+
+	marshaled, err = json.Marshal(unpacker.(map[string]interface{})["defenderSettings"])
```
Comment (Melchizedech) [inline sdk/settings/defender.go:34]: this one is the only "ugly thing" I did; yet I don't know how to make it better.

there is no endpoint to fetch only "defenderSettings" params.
so I just retrieve the value, then use a unmarshal/marshal to convert it to my defenderSetting struct.

File: sdk/waas/network.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+	"github.com/Hivebrite/twistlock-go/sdk"
+)
+
+type NetworkWaas struct {
+	ID               string        `json:"_id"`
+	Modified         time.Time     `json:"modified"`
+	Owner            string        `json:"owner"`
+	HostEnabled      bool          `json:"hostEnabled"`
+	HostRules        []interface{} `json:"hostRules"`
```
Comment (Melchizedech) [inline sdk/waas/network.go:14]: not implemented

File: sdk/waas/network.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	ContainerEnabled bool          `json:"containerEnabled"`
+	ContainerRules   []interface{} `json:"containerRules"`
```
Comment (Melchizedech) [inline sdk/waas/network.go:16]: not implemented

File: twistlock/resource_admission_policies.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+import (
+	"log"
+
+	"github.com/Hivebrite/twistlock-go/sdk"
+	"github.com/Hivebrite/twistlock-go/sdk/policies"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+var admissionEffects = []string{Allow, Block, Alert}
```
[the remark this answers is not in the document]
Comment (Melchizedech) [reply twistlock/resource_admission_policies.go:12]: const array are not a thing in go : https://stackoverflow.com/questions/13137463/declare-a-constant-array

**Review by Dudesons — APPROVED** [review]

**Pull request merged**